### ISSUE TITLE
False positive type mismatches in rustc_codegen_llvm

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rust-analyzer#21621]
**rust-analyzer version**: rust-analyzer version: 0.4.2787-standalone [/home/gh-bjorn3/.vscode-server/extensions/rust-lang.rust-analyzer-0.4.2787-linux-arm64/server/rust-analyzer]

**rustc version**: `rustc 1.94.0-beta.1 (9b1f8ff42 2026-01-19)`

**editor or extension**: VSCode

**relevant settings**: Fairly standard config for the rust repo

**repository link (if public, optional)**: https://github.com/rust-lang/rust/blob/d34f1f931489618efffc4007e6b6bdb9e10f6467/compiler/rustc_codegen_llvm/src/llvm/ffi.rs#L1129

**code snippet to reproduce**:
for example:
```rust
extern "C"  {
    pub(crate) fn LLVMPositionBuilderAtEnd<'a>(Builder: &Builder<'a>, Block: &'a BasicBlock);
}
```

```
expected Builder<'a>, found fn Builder<'_>(InvariantOpaque<'_>) -> Builder<'_> rust-analyzer(E0308)
```

**Comment by ChayimFriedman2** [comment]
Is that all code necessary to reproduce?

**Comment by bjorn3** [comment]
`Builder` also needs to be defined. Updated with a minimal reproducer.

**Comment by ChayimFriedman2** [comment]
Root cause: we treat parameters to extern fns as patterns, while unlike ordinary functions patterns are not allowed there and they should be treated as identifiers.

**Comment by bjorn3** [comment]
Can confirm this is now fixed. Thanks!
